contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+    function setPayout(address[] memory __recipientIds, uint256[] memory _amounts)
+        external
+        onlyPoolManager(msg.sender)
+        onlyAfterAllocation
+    {
+        uint256 recipientLength = __recipientIds.length;
+        if (recipientLength != _amounts.length) {
+            revert INVALID();
+        }
+
+        for (uint256 i = 0; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+
+            payoutSummaries[recipientId] = PayoutSummary(_recipients[recipientId].recipientAddress, _amounts[i]);
+
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Claim allocated tokens
+    /// @param _claims Claims to be claimed
+    function claim(Claim[] calldata _claims) external nonReentrant onlyAfterAllocation {
+        uint256 claimsLength = _claims.length;
+        for (uint256 i; i < claimsLength;) {
+            Claim memory singleClaim = _claims[i];
+            Recipient memory recipient = _recipients[singleClaim.recipientId];
+            uint256 amount = claims[singleClaim.recipientId][singleClaim.token];
+
+            claims[singleClaim.recipientId][singleClaim.token] = 0;
+            _transferAmount(singleClaim.token, recipient.recipientAddress, amount);
+
+            emit Claimed(singleClaim.recipientId, recipient.recipientAddress, amount, singleClaim.token);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Withdraw funds from pool
+    /// @param _amount The amount to be withdrawn
+    function withdraw(uint256 _amount) external onlyPoolManager(msg.sender) {
+        if (block.timestamp <= allocationEndTime + 30 days) {
+            revert NOT_ALLOWED();
+        }
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        if (pool.amount < _amount) {
+            revert UNAUTHORIZED();
+        }
+
+        // TODO: This can cause pool owner stealing funds
+
+        allo.decreasePoolTotalFunding(poolId, _amount);
+        _transferAmount(pool.token, msg.sender, _amount);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Returns status of the pool
+    function _isPoolActive() internal view override returns (bool) {
+        if (registrationStartTime <= block.timestamp && block.timestamp <= registrationEndTime) {
+            return true;
+        }
+        return false;
+    }
+
+    /// @notice Submit application to pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyActiveRegistration
+        returns (address recipientId)
+    {
+        if (_recipients[recipientId].recipientStatus != InternalRecipientStatus.Accepted) {
+            revert RECIPIENT_ALREADY_ACCEPTED();
```

Comment:
this if clause can be removed, or?

Reply:
that should be `  if (_recipients[recipientId].recipientStatus == InternalRecipientStatus.Accepted) {`
we don't do allow re-registration on accepted application right ?

Reply:
we do
they will be set to pending again